ui/src/features/Canvas/index.tsx

```diff
@@ -85,6 +87,34 @@ const Canvas: React.FC<Props> = ({
     onNodePickerOpen,
   });
 
+  const [menuPosition, setMenuPosition] = useState<MenuPosition | null>(null);
+
+  const [selectedNode, setSelectedNode] = useState<Node | null>(null);
```

Comment:
We don't need this. Just do the same as the example where you add the id the the `menu` state as like `nodeContextMenuMeta` or something.